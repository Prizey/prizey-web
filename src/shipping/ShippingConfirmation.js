import React from 'react'
import { connect } from 'react-redux'
import { Redirect } from '@reach/router'
import { withStyles } from '@material-ui/core/styles'
import { Button, Typography } from '@material-ui/core'

import { clearProduct } from 'store/basket/actions'

import GoBack from 'design/GoBack/GoBack'
import Layout from 'design/Layout/Layout'

const styles = theme => ({
  productImage: {
    display: 'block',
    margin: 'auto',
    marginBottom: theme.spacing.md,
    marginTop: theme.spacing.md,
    width: '200px',
  },
})

const ShippingComponent = withStyles(styles)(
  ({ product, classes, difficulty, navigate, ...props }) =>
    product ? (
      <Layout leftIcon={<GoBack to="/" />}>
        <Typography align="center" variant="h5">
          Sweet!
        </Typography>

        <img
          alt={product.title}
          src={product.image}
          className={classes.productImage}
        />

        <Typography align="center">
          We&apos;ll confirm through email.
        </Typography>

        <Button
          variant="contained"
          color="primary"
          fullWidth
          aria-label="Play again"
          onClick={() => {
            props.clearProduct()
            navigate('/')
          }}
        >
          PLAY AGAIN
        </Button>
      </Layout>
    ) : (
      <Redirect to="/" noThrow />
    ),
)

export const mapState = ({ basket = {} }) => ({
  product: basket.product,
})

const ShippingConfirmation = connect(
  mapState,
  { clearProduct },
)(props => <ShippingComponent {...props} />)

export default ({ currentUser, ...props }) =>
  currentUser ? (
    <ShippingConfirmation {...props} />
  ) : (
    <Redirect to="/sign-in?next=/shipping-confirmation" noThrow />
  )
