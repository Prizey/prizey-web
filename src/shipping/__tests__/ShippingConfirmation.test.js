import React from 'react'
import renderer from 'react-test-renderer'
import ShippingConfirmation, { mapState } from '../ShippingConfirmation'

jest.mock('react-redux', () => ({
  connect: () => Component => props => (
    <Component
      {...props}
      clearProduct={jest.fn()}
      product={{ image: '/mock/trump-mask.png', title: 'Trump mask' }}
    />
  ),
}))

jest.mock('croods', () => ({
  New: props => (
    <div {...props}>
      New - <div>render - {props.render(props)}</div>
    </div>
  ),
}))

it('renders correctly', () => {
  const tree = renderer
    .create(<ShippingConfirmation currentUser difficulty="easy" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('redirects correctly', () => {
  const tree = renderer.create(<ShippingConfirmation />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('map the state to props', () => {
  const basket = {
    product: { image: '/mock/trump-mask.png', title: 'Trump mask' },
  }

  expect(mapState({ basket })).toEqual({ product: basket.product })
})

it('trigger the alert', () => {
  const navigate = jest.fn()
  const tree = renderer.create(
    <ShippingConfirmation currentUser navigate={navigate} />,
  ).root

  tree.findByProps({ 'aria-label': 'Play again' }).props.onClick()
  expect(navigate).toHaveBeenCalledTimes(1)
})
