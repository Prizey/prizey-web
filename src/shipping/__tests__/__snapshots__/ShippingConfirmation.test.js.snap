// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redirects correctly 1`] = `null`;

exports[`renders correctly 1`] = `
<div
  className="Component-root-6"
>
  <div
    className="Component-header-5"
  >
    <a
      aria-current="page"
      aria-label="Go Back"
      className="MuiButtonBase-root-14 MuiIconButton-root-8 Component-root-7"
      href="/"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiIconButton-label-13"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-17 MuiSvgIcon-fontSizeLarge-25"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-88"
      />
    </a>
  </div>
  <div
    className="Component-container-4"
  >
    <h2
      className="MuiTypography-root-26 MuiTypography-h2-39 MuiTypography-alignCenter-49"
    >
      PRIZEY
    </h2>
    <div
      className="Component-caption-2"
    />
    <div
      className="Component-children-3"
    >
      <h5
        className="MuiTypography-root-26 MuiTypography-h5-42 MuiTypography-alignCenter-49"
      >
        Sweet!
      </h5>
      <img
        alt="Trump mask"
        className="Component-productImage-1"
        src="/mock/trump-mask.png"
      />
      <p
        className="MuiTypography-root-26 MuiTypography-body1-35 MuiTypography-alignCenter-49"
      >
        We'll confirm through email.
      </p>
      <button
        aria-label="Play again"
        className="MuiButtonBase-root-14 MuiButton-root-62 MuiButton-contained-73 MuiButton-containedPrimary-74 MuiButton-raised-76 MuiButton-raisedPrimary-77 MuiButton-fullWidth-87"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-63"
        >
          PLAY AGAIN
        </span>
        <span
          className="MuiTouchRipple-root-88"
        />
      </button>
    </div>
    <p
      className="MuiTypography-root-26 MuiTypography-body1-35 MuiTypography-alignCenter-49"
    />
  </div>
</div>
`;
